test/rooms_test.js

```diff
@@ -1538,6 +1549,96 @@ describe("/rooms", function() {
+          if (err) throw err;
+          deleteRooms(hawkCredentials, [roomToken]).end(function(err, res) {
+            if (err) throw err;
+            storage.getRoomParticipants(roomToken, function(err, participants) {
+              if (err) throw err;
+              expect(participants).to.length(0);
+              done();
+            });
+          });
+        });
+      });
+    });
+
+    it("should delete rooms if the user is the room owner", function(done) {
+      createRoom(hawkCredentials).end(function(err, res) {
+        if (err) throw err;
+        var roomToken = res.body.roomToken;
+        createRoom(hawkCredentials).end(function(err, res) {
+          if (err) throw err;
+          var roomToken2 = res.body.roomToken;
+          requests = [];
+          deleteRooms(hawkCredentials, [roomToken, roomToken2]).end(function(err, res) {
+            if (err) throw err;
+            expect(requests).to.length(1);
+            getRoomInfo(hawkCredentials, roomToken, 404).end(function(err) {
+              if (err) throw err;
+              getRoomInfo(hawkCredentials, roomToken2, 404).end(done);
+            });
+          });
+        });
+      });
+    });
+
+    it("should return a 404 if no room where found", function(done) {
+      createRoom(hawkCredentials).end(function(err, res) {
+        if (err) throw err;
+        var roomToken = res.body.roomToken;
+        createRoom(hawkCredentials).end(function(err, res) {
+          if (err) throw err;
+          var roomToken2 = res.body.roomToken;
+          deleteRooms(hawkCredentials, [roomToken, roomToken2]).end(function(err, res) {
+            if (err) throw err;
+            var data = {responses: {}};
+            data.responses[roomToken] = {code: 200};
+            data.responses[roomToken2] = {code: 200};
+            expect(res.body).to.eql(data);
+
+            deleteRooms(hawkCredentials, [roomToken, roomToken2], 404).end(function(err, res2) {
+              if (err) throw err;
+              var data = {responses: {}};
+              data.responses[roomToken] = {code: 404, errno: 105, error: "Room not found."};
+              data.responses[roomToken2] = {code: 404, errno: 105, error: "Room not found."};
+              expect(res2.body).to.eql(data);
+              done();
+            });
+          });
+        });
+      });
+    });
+
+    it("should return a 207 if some room where found and other not", function(done) {
+      createRoom(hawkCredentials).end(function(err, res) {
+        if (err) throw err;
+        var roomToken = res.body.roomToken;
+        deleteRooms(hawkCredentials, [roomToken, "foobar"], 207).end(function(err, res) {
+          if (err) throw err;
+          var data = {responses: {
+            "foobar": {code: 404, errno: 105, error: "Room not found."}
+          }};
+          data.responses[roomToken] = {code: 200};
+          expect(res.body).to.eql(data);
+          done();
+        });
+      });
+    });
+
+  });
```

Comment:
We should also make sure we return a 503 in case an error occured when deleting everything.